### ISSUE TITLE
Support a way to ignore found issues in the future.

### ISSUE THREAD

**Issue by TheRealAgentK** [cflint/CFLint#195]
Let's say I in general want to run the rule that checks for missing default values in non-required function arguments.

But I might have a few custom instances for which I don't want this rule to be applied - maybe because it's been taken care of in the code. So we'd want to exclude a certain function from triggering it.

As far as I can see there's no way to currently do that. Is there any option to implement something like this with Annotations, similarly to how Java linters do offer that (for instance in Android linting). I'd annotate a function so that it ignores a specific rule for future runs.

**Comment by bardware** [comment]
I know how an empty log file in the search for errors feels but I believe it's deceptive to disable checks.
If you for instance log CFLint output as JSON and import in somwehere you might exclude results based on the data provided. filename and function name are output in most cases.
If you look at the raw output you would still see there's an issue - whether it's bad/important or not.

**Comment by TheRealAgentK** [comment]
Sure, I could build all sorts of solutions on top of the cflint output and manually filtering out results I don't want. 

But having a way to annotate parts of ones code with instructions to not check and apply certain rules is really the norm. The fact that you find it deceptive to disable checks is appreciated, but you wouldn't have to use that feature if you don't like it. The option should be there nevertheless.

**Comment by ryaneberly** [comment]
@TheRealAgentK , I hear you.  We haven't supported code annotations yet.  But you can accomplish some of this with the cflintexclude.json file
https://github.com/cflint/CFLint/wiki/Are-There-Any-Recipes-Out-There%3F

You can filter out codes by file, function name, and line number.

**Comment by ryaneberly** [comment]
Added support for 
//cflint ignore:line
which suppresses all messages for that line.

and 
//cflint ignore:MISSING_VAR
which suppresses a comma separated list of message codes for that line.

**Comment by ryaneberly** [comment]
What syntax would you suggest for function annotations?

something like:
/*
@CFLintIgnore MISSING_VAR
*/
function somefunction(){
    myVar = 123;
}

**Comment by TheRealAgentK** [comment]
Yes, I like that. 

In Android for instance, it'd be:
...
@SuppressLint("NewApi")
public void onBackStackChanged() { ... }
...

This syntax would obviously not work in CF, so it'd have to be in a comment as you've shown it. 

One question: What would the syntax look like if I wanted to suppress multiple lint rules for a function? 

In Android that'd work via:

@SuppressLint({"NewApi","StringFormatInvalid"}), but I find this syntax rather clunky and convoluted

Going with your example, I'd like to be able to just use a comma-separated list:

@CFLintIgnore MISSING_VAR,SOMETHINGELSE,ANOTHERTHINGTOIGNORE

**Comment by ryaneberly** [comment]
Support added at the line level in cfscript:
` someVar = 123; // cflint ignore:line`

at the component and function level in cfscript:
```
	/*
	other comments....
	@CFLintIgnore SOMETHINGELSE,MISSING_VAR,ANOTHERTHINGTOIGNORE
	*/
         component{
      ......

```
at the component and function level in CFML:
```
       <!---
       	other comments....
       @CFLintIgnore SOMETHINGELSE,MISSING_VAR,ANOTHERTHINGTOIGNORE
       --->
         <cfcomponent>
          .....
```

**Comment by TheRealAgentK** [comment]
Is there any support for lines in normal tag-based code outside ignoring a whole function or a whole component?

**Comment by ryaneberly** [comment]
Yes @TheRealAgentK , 

*only* at the component and function level in CFML:
```
       <!---
       	other comments....
       @CFLintIgnore SOMETHINGELSE,MISSING_VAR,ANOTHERTHINGTOIGNORE
       --->
         <cfcomponent>
          .....
```

I think that's working.  There's a test here:
https://github.com/cflint/CFLint/blob/cc45d5646dca6dd1a3a89854bead5b60bc6eebbd/src/test/resources/com/cflint/tests/Ignores/ignoreCFMLComponent2.cfc

**Comment by ryaneberly** [comment]
the project is sadly in need of some documentation....

**Comment by TheRealAgentK** [comment]
The current mechanism doesn't seem to work across all rules.

In my case I can't get CFQUERYPARAM_REQ to be ignored at all. Added test case demonstrating the issue to dev (https://github.com/cflint/CFLint/commit/c88b366a23e53ac8b4bc2079fbbb3d864c2ae6ec).

I'm not sure if there are any other places I could possibly put the @CFLintIgnore annotation :)

I noticed that all the test cases were for MISSING_VAR only - is there some dependency on a certain category of rules in the implementation of this?
